packages/back-end/src/controllers/experiments.ts

```diff
@@ -873,8 +873,10 @@ export async function postExperiment(
     "excludeFromPayload",
   ] as (keyof ExperimentInterfaceStringDates)[]).some((key) => key in changes);
   if (needsRunExperimentsPermission) {
+    const linkedFeatures = await getLinkedFeatureInfo(context, experiment);
```

Comment:
`getLinkedFeatureInfo` is a very expensive function and overkill for what we need.  It fetches all revisions of all linked features when we only care about the rules in the FeatureInterface itself.  There are some users with features that have thousands of revisions so if we can avoid querying them, that's ideal.

Reply:
@jdorn Given this, it feels like there could be a much simpler solution here.

We could update the signature of `getAffectedEnvsForExperiment` to take in the experiment and the linkedFeatures, but the shape of the linkedFeatures is just the `FeatureInterface[]`.

E.G.

```ts
export function getAffectedEnvsForExperiment({
  experiment,
  linkedFeatures,
}: {
  experiment: ExperimentInterface | ExperimentInterfaceStringDates;
  linkedFeatures?: FeatureInterface[];
}): string[] {
  // Visual changesets are not environment-scoped, so it affects all of them
  // Also fallback to all envs if linkedFeatures is undefined, but the experiment does actually have linked features
  if (
    experiment.hasVisualChangesets ||
    experiment.hasURLRedirects ||
    (!linkedFeatures && !!experiment.linkedFeatures?.length)
  )
    return ["__ALL__"];

  if (linkedFeatures?.length) {
    const envs: string[] = [];

    linkedFeatures.forEach((linkedFeature) => {
      const envSettings = linkedFeature.environmentSettings;
      if (!linkedFeature.archived) {
        for (const env in envSettings) {
          if (envSettings[env].enabled) {
            envs.push(env);
          }
        }
      }
    });
   return envs;
  }

  return [];
}
```